src/solvers/algebraic_solver.hpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file algebraic_solver.hpp
+ *
+ * @brief This file contains the declaration of an iterative solver wrapper
+ */
+
+#ifndef ALGEBRAIC_SOLVER
+#define ALGEBRAIC_SOLVER
+
+#include <memory>
+#include <optional>
+#include <variant>
+
+#include "common/serac_types.hpp"
+#include "mfem.hpp"
+
+namespace serac {
+
+/**
+ * Wraps a (currently iterative) system solver and handles the configuration of linear
```

Comment:
I would add a comment that this solves a generic global system of nonlinear algebraic equations.